bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BTable.java

```diff
@@ -186,7 +188,11 @@ public void reset(boolean isInTransaction) {
         // Make next row the current row
         moveToNext();
         // Create BStruct from current row
-        return (BMap<String, BValue>) iterator.generateNext();
+        if (iterator != null) {
+            return (BMap<String, BValue>) iterator.generateNext();
+        } else {
+            return new BMap<>(BTypes.typeAny);
```

Comment:
We can move out the return statement here and remove the else block right? Same at https://github.com/ballerina-platform/ballerina-lang/pull/10352/files#diff-782094a79e6831ac57a34ad157af49c1R462 ?